tests/lib/Creator/DataSourceTestItemCreator.php

```diff
@@ -0,0 +1,228 @@
+use EzSystems\EzRecommendationClient\Tests\Stubs\ItemList;
+use EzSystems\EzRecommendationClient\Tests\Stubs\ItemType;
+use Ibexa\Contracts\Personalization\Criteria\CriteriaInterface;
+use Ibexa\Contracts\Personalization\Value\ItemInterface;
+use Ibexa\Contracts\Personalization\Value\ItemListInterface;
+use Ibexa\Contracts\Personalization\Value\ItemTypeInterface;
+use Traversable;
+
+final class DataSourceTestItemCreator
+{
+    /**
+     * @phpstan-param ?iterable<string, array{
+     *  'item_type_identifier': string,
+     *  'item_type_name': string,
+     *  'languages': array<string>,
+     *  'limit': int,
+     * }> $itemsConfig
+     *
+     * @return Traversable<\Ibexa\Contracts\Personalization\Value\ItemInterface>
+     */
+    public static function createTestItems(?iterable $itemsConfig = null): Traversable
+    {
+        return self::createTestItemsForConfig(
+            $itemsConfig ?? self::getDefaultItemsConfig()
+        );
+    }
+
+    public static function createTestItem(
+        int $counter,
+        string $itemId,
+        string $itemTypeIdentifier,
+        string $itemTypeName,
+        string $language
+    ): ItemInterface {
+        return new Item(
+            $itemId,
+            self::createTestItemType($itemTypeIdentifier, $itemTypeName),
+            $language,
+            self::createTestItemAttributes(
+                $counter,
+                $itemTypeIdentifier,
+                $itemTypeName,
+                $language
+            )
+        );
+    }
+
+    public static function createTestItemType(string $identifier, string $name): ItemTypeInterface
+    {
+        return new ItemType(
+            $identifier,
+            $name
+        );
+    }
+
+    /**
+     * @return array<string, string|int|array>
+     */
+    public static function createTestItemAttributes(
+        int $counter,
+        string $itemTypeIdentifier,
+        string $itemTypeName,
+        string $language
+    ): array {
+        return [
+            'name' => sprintf('%s %s %s', $itemTypeName, $counter, $language),
+            'body' => sprintf('%s %s %s %s', $itemTypeName, $counter, Item::ITEM_BODY, $language),
+            'image' => sprintf(Item::ITEM_IMAGE, $itemTypeIdentifier, $counter),
+        ];
+    }
+
+    /**
+     * @param array<string> $identifiers
+     * @param array<string> $languages
+     */
+    public static function createTestCriteria(
+        array $identifiers,
+        array $languages,
+        int $limit = Criteria::LIMIT,
+        int $offset = 0
+    ): CriteriaInterface {
+        return new Criteria(
+            $identifiers,
+            $languages,
+            $limit,
+            $offset
+        );
+    }
+
+    public static function createTestEmptyItemList(): ItemListInterface
+    {
+        return new ItemList([]);
+    }
+
+    public static function createTestItemList(ItemInterface ...$items): ItemListInterface
+    {
+        return new ItemList($items);
+    }
```

Comment:
Those two can be combined. If variadic argument is not provided, an empty array takes it's place.

```suggestion
    public static function createTestItemList(ItemInterface ...$items): ItemListInterface
    {
        return new ItemList($items);
    }
```